libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3148,6 +3148,16 @@ void TypeChecker::endVisit(Literal const& _literal)
 	_literal.annotation().isPure = true;
 }
 
+void TypeChecker::endVisit(UsingForDirective const& _usingFor)
+{
+	if (m_currentContract->isInterface())
+		m_errorReporter.typeError(
+			9088_error,
+			_usingFor.location(),
+			"The \"using for\" directive is not allowed for interfaces."
```

Comment:
```suggestion
			"The \"using for\" directive is not allowed inside interfaces."
```
(otherwise it might be read that `L` in `using L for *` cannot be an interface.)